internal/sync/sync.go

```diff
@@ -91,74 +91,70 @@ func (s Sync) syncPaths(delete bool, local, remote string) error {
 	conn, err := s.Client.DialWsep(ctx, &s.Env)
 	if err != nil {
-		return err
+		return xerrors.Errorf("dial websocket: %w", err)
 	}
-	defer conn.Close(websocket.CloseNormalClosure, "")
+	defer func() { _ = conn.Close(websocket.CloseNormalClosure, "") }() // Best effort.
 
 	execer := wsep.RemoteExecer(conn)
 	process, err := execer.Start(ctx, wsep.Command{
 		Command: prog,
 		Args:    args,
 	})
 	if err != nil {
-		return err
+		return xerrors.Errorf("exec remote process: %w", err)
 	}
-	go io.Copy(os.Stdout, process.Stderr())
-	go io.Copy(os.Stderr, process.Stdout())
+	// TODO: Consider checking the copy errors.
+	go func() { _, _ = io.Copy(os.Stdout, process.Stdout()) }() // Best effort.
+	go func() { _, _ = io.Copy(os.Stderr, process.Stderr()) }() // Best effort.
 
-	err = process.Wait()
-	if code, ok := err.(wsep.ExitError); ok {
-		return fmt.Errorf("%s exit status: %v", prog, code)
-	}
-	if err != nil {
+	if err = process.Wait(); err != nil {
+		if code, ok := err.(wsep.ExitError); ok {
+			return fmt.Errorf("%s exit status: %d", prog, code)
+		}
 		return xerrors.Errorf("execution failure: %w", err)
 	}
+
 	return nil
 }
 
 // initSync performs the initial synchronization of the directory.
 func (s Sync) initSync() error {
-	flog.Info("doing initial sync (%v -> %v)", s.LocalDir, s.RemoteDir)
+	flog.Info("doing initial sync (%s -> %d)", s.LocalDir, s.RemoteDir)
 
 	start := time.Now()
 	// Delete old files on initial sync (e.g git checkout).
 	// Add the "/." to the local directory so rsync doesn't try to place the directory
 	// into the remote dir.
-	err := s.syncPaths(true, s.LocalDir+"/.", s.RemoteDir)
-	if err == nil {
-		flog.Success("finished initial sync (%v)", time.Since(start).Truncate(time.Millisecond))
+	if err := s.syncPaths(true, s.LocalDir+"/.", s.RemoteDir); err != nil {
+		return err
 	}
-	return err
+	flog.Success("finished initial sync (%v)", time.Since(start).Truncate(time.Millisecond))
+	return nil
 }
 
 func (s Sync) convertPath(local string) string {
 	relLocalPath, err := filepath.Rel(s.LocalDir, local)
 	if err != nil {
 		panic(err)
 	}
-	return filepath.Join(
-		s.RemoteDir,
-		relLocalPath,
-	)
+	return filepath.Join(s.RemoteDir, relLocalPath)
 }
 
 func (s Sync) handleCreate(localPath string) error {
 	target := s.convertPath(localPath)
-	err := s.syncPaths(false, localPath, target)
-	if err != nil {
-		_, statErr := os.Stat(localPath)
+
+	if err := s.syncPaths(false, localPath, target); err != nil {
 		// File was quickly deleted.
-		if os.IsNotExist(statErr) {
+		if _, e1 := os.Stat(localPath); os.IsNotExist(e1) { // NOTE: Discard any other stat error and just expose the syncPath one.
 			return nil
 		}
-
 		return err
 	}
 	return nil
 }
 
 func (s Sync) handleDelete(localPath string) error {
-	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
```

Comment:
I wouldn't multiply by 1?

Reply:
for consistency and readability, I think it is best to always specify the full time, it shows it is intended and not an oversight. As it is a const, this is handled at compile time and have no impact and the runtime, it is purely aesthetic.

With this in mind, if you think it is best to remove it, we sure can.

Reply:
I'm still for removing, in a "less is more" sort of approach.